### PR TITLE
fix(ivy): allow inserting attached views to a different container

### LINKED ISSUE #34152: ivy: failed to create dynamic component
# 🐞 bug report

### Affected Package
The issue is caused by package @angular/core

### Is this a regression?

Yes, it worked fine in v8

### Description

Trying to create a dynamic component fails to work when calling `insert` on a `ViewContainerRef` with a component instance's `hostView`

## 🔬 Minimal Reproduction
```
git clone https://github.com/mattlewis92/ivy-dynamic-component-bug.git
cd ivy-dynamic-component-bug
npm i
npm start -- --open
```

then click the "click me" button and you'll see the error thrown

## 🔥 Exception or Error
<pre><code>
ERROR Error: ASSERTION ERROR: index must be positive
    at throwError (core.js:1229)
    at assertGreaterThan (core.js:1213)
    at ViewContainerRef_._adjustIndex (core.js:17341)
    at ViewContainerRef_.insert (core.js:17264)
    at ViewContainerRef_.move (core.js:17288)
    at ViewContainerRef_.insert (core.js:17264)
    at SafeSubscriber._next (modal-keeper.component.ts:24)
    at SafeSubscriber.__tryOrUnsub (Subscriber.js:183)
    at SafeSubscriber.next (Subscriber.js:122)
</code></pre>

## 🌍  Your Environment

**Angular Version:**
<pre><code>
Angular CLI: 9.0.0-rc.4
Node: 12.10.0
OS: darwin x64

Angular: 9.0.0-rc.4
... animations, cli, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.900.0-rc.4
@angular-devkit/build-angular     0.900.0-rc.4
@angular-devkit/build-optimizer   0.900.0-rc.4
@angular-devkit/build-webpack     0.900.0-rc.4
@angular-devkit/core              9.0.0-rc.4
@angular-devkit/schematics        9.0.0-rc.4
@ngtools/webpack                  9.0.0-rc.4
@schematics/angular               9.0.0-rc.4
@schematics/update                0.900.0-rc.4
rxjs                              6.5.3
typescript                        3.6.4
webpack                           4.41.2
</code></pre>

**Anything else relevant?**
nope

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**
This PR fixes #34152
Easier to review commit-by-commit.

### CODE REVIEW

**Review by mhevery — APPROVED** [review]

File: packages/core/src/render3/view_engine_compatibility.ts [hunk 1]
```diff
@@ -240,18 +240,40 @@ export function createContainerRef(
       }
 
       insert(viewRef: viewEngine_ViewRef, index?: number): viewEngine_ViewRef {
+        const lView = (viewRef as ViewRef<any>)._lView !;
+
         if (viewRef.destroyed) {
-          throw new Error('Cannot insert a destroyed View in a ViewContainer!');
+          throw new Error(`Cannot insert a destroyed View in a ViewContainer!`);
```
Comment (AndrewKushnir) [inline packages/core/src/render3/view_engine_compatibility.ts]: @pkozlowski-opensource just curious why we need backticks here vs single quote? If I understand correctly, backticks will require down-leveling in ES5.

Comment (pkozlowski-opensource) [reply packages/core/src/render3/view_engine_compatibility.ts]: @AndrewKushnir oh, this is a left-over after some refactoring I've tried. We don't need it, I will revert. Thnx for catching!

**Comment by mhevery** [comment]
[presubmit](http://test/OCL:283566724:BASE:283566757:1575396058430:c1858285)

File: packages/core/src/render3/view_engine_compatibility.ts (hunk 1 above)
Comment (AndrewKushnir) [reply packages/core/src/render3/view_engine_compatibility.ts]: Thanks Pawel!

**Comment by mhevery** [comment]
This PR seems to break the `g3`. The breakage is a screenshoot diff where the UI is missing most of its content, so it seems to be related to this change.

**Comment by mhevery** [comment]
@AndrewKushnir thinks they are flakes, we are re-running it here: https://test.corp.google.com/ui#id=OCL:283566724:BASE:283620405:1575411379488:e72ec0a8

**Pull request closed without merging**